src/http/client/http.test.ts

```diff
@@ -0,0 +1,218 @@
+            'Content-Length': Buffer.byteLength(postData)
+          }
+        },
+        expect.anything()
+      )
+
+      expect(mockReq.write).toHaveBeenCalledWith(postData)
+      expect(mockReq.end).toHaveBeenCalled()
+    })
+  })
+
+  describe('request callback', () => {
+    it('should call request callback successfully', async () => {
+      // @ts-ignore
+      const mockRes: IncomingMessage = {
+        ...baseMockRes
+      }
+
+      const resolve = jest.fn()
+      const reject = jest.fn()
+      const resData = { key: 'value' }
+
+      // @ts-ignore
+      mockRes.on.mockImplementation((event: any, cb: any): any => {
+        if (event === 'end') {
+          cb()
+        } else if (event === 'data') {
+          cb(JSON.stringify(resData))
+        }
+      })
+
+      // @ts-ignore
+      httpClient.requestCb(mockRes, resolve, reject)
+
+      // @ts-ignore
+      expect(mockRes.on).toHaveBeenCalledWith('data', expect.any(Function))
+      expect(mockRes.on).toHaveBeenCalledWith('end', expect.any(Function))
+      expect(mockRes.setEncoding).toHaveBeenCalledWith('utf8')
+      expect(resolve).toHaveBeenCalledWith({
+        headers: mockRes.headers,
+        data: resData,
+        status: mockRes.statusCode
+      })
+    })
+
+    it('should call request callback with http error', async () => {
+      // @ts-ignore
+      const mockRes: IncomingMessage = {
+        ...baseMockRes,
+        statusCode: 500
+      }
+
+      const resolve = jest.fn()
+      const reject = jest.fn()
+      const resData = { key: 'value' }
+
+      // @ts-ignore
+      mockRes.on.mockImplementation((event: any, cb: any): any => {
+        if (event === 'end') {
+          cb()
+        } else if (event === 'data') {
+          cb(JSON.stringify(resData))
+        }
+      })
+
+      // @ts-ignore
+      httpClient.requestCb(mockRes, resolve, reject)
+
+      // @ts-ignore
+      expect(mockRes.on).toHaveBeenCalledWith('data', expect.any(Function))
+      expect(mockRes.on).toHaveBeenCalledWith('end', expect.any(Function))
+      expect(mockRes.setEncoding).toHaveBeenCalledWith('utf8')
+      expect(resolve).toHaveBeenCalledWith({
+        headers: mockRes.headers,
+        err: expect.any(Error),
+        status: mockRes.statusCode
+      })
+    })
+  })
+
+  it('should call request callback with invalid headers', async () => {
+    // @ts-ignore
+    const mockRes: IncomingMessage = {
+      ...baseMockRes,
+      headers: {
+        'content-type': 'application/x-javascript'
+      }
+    }
+
+    const resolve = jest.fn()
+    const reject = jest.fn()
+
+    // @ts-ignore
+    mockRes.on.mockImplementation((event: any, cb: any): any => {
+      if (event === 'end') {
+        cb()
+      }
+    })
+
+    // @ts-ignore
+    httpClient.requestCb(mockRes, resolve, reject)
+
+    // @ts-ignore
+    expect(mockRes.on).toHaveBeenCalledWith('data', expect.any(Function))
+    expect(mockRes.on).toHaveBeenCalledWith('end', expect.any(Function))
```

Comment:
Wasn't the issue that you are not testing the function ? Since you `expect.any`. Or is it tested in some other way?

Reply:
The magic happen here:
```
mockRes.on.mockImplementation((event: any, cb: any): any => {
        if (event === 'end') {
          cb()
        } else if (event === 'data') {
          cb(JSON.stringify(resData))
        }
      })
```

Then you can validate in your resolve() spy this:
```
expect(resolve).toHaveBeenCalledWith({
        headers: mockRes.headers,
        err: expect.any(Error),
        status: mockRes.statusCode
      })
```

Reply:
Although this will be update why the usage of axios